src/model/wallet.js

```diff
@@ -314,25 +314,42 @@ export const createWallet = async (seed = null, color = null, name = null) => {
     const allWalletsResult = await getAllWallets();
     const allWallets = get(allWalletsResult, 'wallets', {});
 
+    let existingWalletId = null;
+    let isVisible = false;
     if (isImported) {
       // Checking if the generated account already exists
-      const alreadyExisting = Object.keys(allWallets).some(key => {
-        const someWallet = allWallets[key];
-        return someWallet.addresses.some(account => {
-          return (
+      const alreadyExistingWallet = find(allWallets, someWallet => {
+        const existingAccount = find(
+          someWallet.addresses,
+          account =>
             toChecksumAddress(account.address) ===
-              toChecksumAddress(wallet.address) && someWallet.type === type
-          );
-        });
+            toChecksumAddress(wallet.address)
+        );
+        if (existingAccount) {
+          isVisible = !!existingAccount.visible;
+          return true;
+        }
+        return false;
       });
 
-      if (alreadyExisting) {
+      existingWalletId = alreadyExistingWallet?.id;
+
+      // Don't allow adding a readOnly wallet that you have already visible
+      // or a private key that you already have visible as a seed or mnemonic
+      const isPrivateKeyOverwritingSeedMnemonic =
+        type === WalletTypes.privateKey &&
+        (alreadyExistingWallet.type === WalletTypes.seed ||
```

Comment:
it should be `alreadyExistingWallet?.type` otherwise it will throw when there's no already existing wallet bc `null.type`